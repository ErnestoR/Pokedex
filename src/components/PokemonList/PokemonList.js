import React from "react";
import { ListItem, ListItemText } from "@material-ui/core";
import { FixedSizeList } from "react-window";
import { Link } from "react-router-dom";

import useStyles from "./styles";

export function Row(props) {
  const { index, data, style } = props;

  return (
    <ListItem
      button
      style={style}
      key={index}
      divider
      component={Link}
      to={`/pokemon/${data[index].name}`}
    >
      <ListItemText primary={data[index].name} />
    </ListItem>
  );
}

const Home = props => {
  const classes = useStyles();
  const { data } = props;

  return (
    <FixedSizeList
      height={400}
      itemSize={46}
      itemCount={data.length}
      className={classes.list}
      itemData={data}
    >
      {Row}
    </FixedSizeList>
  );
};

export default Home;
