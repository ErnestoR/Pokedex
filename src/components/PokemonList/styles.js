import { makeStyles } from "@material-ui/core/styles";

export default makeStyles(theme => ({
  list: {
    width: "100%",
    backgroundColor: theme.palette.background.paper,
    position: "relative",
    overflow: "auto"
  }
}));
