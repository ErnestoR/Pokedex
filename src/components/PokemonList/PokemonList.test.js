import React from "react";
import { render } from "@testing-library/react";
import PokemonList from "./PokemonList";
import { BrowserRouter as Router } from "react-router-dom";

describe("PokemonList", () => {
  const data = [{ name: "pikachu" }];

  it("renders empty list", () => {
    const { container } = render(<PokemonList data={[]} />);
    expect(container).toMatchSnapshot();
  });

  it("renders list", () => {
    const { container, getByText } = render(
      <Router>
        <PokemonList data={data} />
      </Router>
    );
    expect(getByText("pikachu")).toBeInTheDocument();

    const link = container.querySelector("a");
    expect(link).toHaveAttribute("href", "/pokemon/pikachu");
  });
});
