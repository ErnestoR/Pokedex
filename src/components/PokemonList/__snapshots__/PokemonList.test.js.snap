// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonList renders empty list 1`] = `
<div>
  <div
    class="makeStyles-list-1"
    style="position: relative; height: 400px; overflow: auto; will-change: transform; direction: ltr;"
  >
    <div
      style="height: 0px; width: 100%;"
    />
  </div>
</div>
`;
